app/utils/logger.js

```diff
@@ -0,0 +1,58 @@
+// app/utils/logger.js
+
+const { createLogger, transports, format } = require('winston');
+const { combine, timestamp, printf, colorize, splat, simple } = format;
+const fs = require('fs');
+const path = require('path');
+require('winston-daily-rotate-file');
+
+// Ensure log directory exists
+const logDirectory = path.join(__dirname, '../../logs');
+fs.existsSync(logDirectory) || fs.mkdirSync(logDirectory);
+
+const logFormat = printf(({ level, message, timestamp, ...metadata }) => {
+    let msg = `${timestamp} [${level}]: ${message}`;
+    if (Object.keys(metadata).length) {
```

Comment:
Using the zero value of int as the judgement condition won't be a good idea due to its implicit ambiguity. 
if (Object.keys(metadata).length === 0) is better, since its explicit, accurate, and easy to understand.